doc/develop/api/design_guidelines.rst

```diff
@@ -103,8 +103,15 @@ The Kconfig flag used to enable the feature should be added to the
 ``PREDEFINED`` variable in :file:`doc/zephyr.doxyfile.in` to ensure the
 conditional API and functions appear in generated documentation.
 
+Subsystem namespaces
+********************
+A subsystem can define its own naming conventions for symbols as long as the scheme doesn't collide with Zephyr naming conventions :ref:`api_conventions.rst`. The naming convention should include a
+namespace prefix (for example, bt\_ for Bluetooth LE, or net\_ for IP). This limits possible
+collisions with symbols defined by an application.
+
 Return Codes
 ************
+Zephyr uses the standard codes in ``errno.h`` for all APIs. As a general rule, 0 indicates success; a negative ``errno.h`` code indicates an error condition.
```

Comment:
There are multiple errno.h in the tree and the picolibc module also defines an errno.h.  Can you add some language to clarify how the errno.h is selected?

Reply:
I don't have much understanding of that, sorry.